### DISCUSSION TITLE
Project Owners

### DISCUSSION THREAD

**Discussion by arneschreuder** [COS-301/graduates#1] [General]
This is the discussion board for and with Project Owners.

**Comment by ThaboJChesane** [comment]
@arneschreuder 
Good afternoon

1. The play book says has tests and contracts down for this week. The project managers would like a bit more clarity.

2. Designers have requested me to ask if they could get a license for Adobe XD for the design aspect for the mining project as many IMY students have a free license.

3. In terms of roles, Since most teams have a lot of roles that won't be required and some have multiple roles some teams are wondering if an individual has already fulfilled 2 roles, it would be unnecessary and an administrative nightmare to all switch roles. My original response was roles do not matter that much but they would still your clarification.

**Reply by arneschreuder** [reply]
Hi @ThaboJChesane my feedback below:

* Tests and contracts. So contracts refer to agreements between each layer about what each layer's API looks like. So what do the interfaces look like in the API layer, the Services layer, the repositories and/or db layer. One can start to write which tests need to be executed for these. For example. Lets say the there is an API for authentication. Then the contract refers to the interface for a login mutation call, tests include what the expected input and output should be. So a bunch of define-it TS outlines based on the "contracts"/interfaces.
* You are going to have to ask @funmilola09 or Stacey. I do not have a say in this regard. I will bring it up to them now, we have a meeting at 15:30.
* Unfortunately everyone has to switch roles. I would suggest that people that have two roles pick a primary for this round. Teams are responsible for delegating work appropriately. If one person is overloaded or underutilised, then that is the PM's problem and it should be voiced to the PM. We will consider both roles a user has filled in sprint 1 and 2. The roles are there to just guide a bit on responsibility separation.

**Reply by arneschreuder** [reply]
However, I will bring it up with Stacey and @funmilola09 on this topic and provide feedback. For now just continue with sprint 1 as said above.

**Reply by ThaboJChesane** [reply]
Thank you I have emailed Ms Stacey twice about it but she has not gotten back to me

**Reply by arneschreuder** [reply]
@ThaboJChesane Ok, so we are in communication about the role switching. There will be a formal communication on this, but we feel that since the students are doing well so far in the mini-project and we will probably drop the requirement to switch roles. If they still want to they are more than welcome, perhaps we will allow a small window for this to happen, but it will probably not be a REQUIREMENT anymore. Just wait for formal communication. 

Further, on the Adobe XD thing, I did talk to Stacey and she will find out for us if we can organise this in a timely manner. In the mean time, one would have to consider that access might not be given in time, since we are 25% into the Mini Project.

**Reply by ThaboJChesane** [reply]
ok thank you

**Comment by ThaboJChesane** [comment]
Good morning @arneschreuder 
Students wanted to know with regards to the mentioned extension(15th to 18th) because of the broken pipeline, when do plan to make the announcement

**Reply by arneschreuder** [reply]
I posted this detail in the project managers discussion. But we will send the email formally this week. The release branch will be created the 18th
